### PR TITLE
Added v4 API swagger changes to the documentation

### PR DESCRIPTION
**Pull request by khoibrandon**

### CODE REVIEW

File: src/api-reference/financial-integration/v4.financial-integration.markdown [hunk 1]
```diff
@@ -141,23 +148,78 @@ Content-Length: 491
...
+  }
+   ],
+  "page" : {
+    "size" : 3,
+    "totalElements" : 8,
+    "totalPages" : 3,
+    "number" : 0
+  }
 }
-```
```
Comment (sarraloew) [inline src/api-reference/financial-integration/v4.financial-integration.markdown]: Put this set of three backticks back in- it makes the example format correctly.

File: src/api-reference/financial-integration/v4.financial-integration.markdown [hunk 2]
```diff
@@ -429,24 +607,44 @@ Code|Description|Category
...
-`docType`|`string`|-|Transaction type. Supported values are: `expense`, `invoice`, or `cashadvance`.
+`docType`|`string`|-|Transaction type. Supported values are: `expense`, `invoice`, `cashadvance`, `payroll`, `obligation`.
 `companyId`|`string`|-|Unique identifier for the company in Concur.
 `entityId`|`string`|-|Unique Identifier for the entity in Concur.
 `companyUuid`|`string`|`UUID`|Unique UUID for the company in Concur.
 `erpSystemId`|`string`|-|The external System ID that processed the document.
 `document`|`Array`|-|The JSON financial document. Review the [FI sample documents](#examples) below
 `docStatus`|`string`|-|The financial document status. Supported values are: `READY`, `ACKNOWLEDGED`, `POSTING_CONFIRMED_SUCCESS`, `POSTING_CONFIRMED_FAILURE`
 
+### <a name="schema-page"></a>Pagination details
```
Comment (sarraloew) [inline src/api-reference/financial-integration/v4.financial-integration.markdown]: Call this schema "PageMetaData"

File: src/api-reference/financial-integration/v4.financial-integration.markdown [hunk 3]
```diff
@@ -429,24 +607,44 @@ Code|Description|Category
 
 Name|Type|Format|Description
 ---|---|---|---
-`fIDocuments`|`Array`|[Financial document](#schema-fidocument)|The result collection.
-`nextPage`|`string`|-|The URI of the next page of results.
-`numberReturned`|`int32`|-|Number of documents returned in current page.
-`offset`|`string`|-|Next page offset.
+`content`|`Array`|[Financial document](#schema-fidocument)|The result collection.
```
Comment (sarraloew) [inline src/api-reference/financial-integration/v4.financial-integration.markdown]: This link text should be FIDocument, if swagger is correct.

File: src/api-reference/financial-integration/v4.financial-integration.markdown [hunk 4]
```diff
@@ -429,24 +607,44 @@ Code|Description|Category
...
+
+Name|Type|Format|Description
+---|---|---|---
+`size`|int32|`-`|Total number of pages returned
+`totalElements`|int32|-|Total count of records returned
+`totalPages`|int32|-|Total number of pages returned
+`number`|int32|-|Page location for which the result is returned, ex: first page starts with number 0, second page starts with 1.
+
+
+### <a name="schema-links"></a>Pagination links
```
Comment (sarraloew) [inline src/api-reference/financial-integration/v4.financial-integration.markdown]: Call this schema just "Link"

File: src/api-reference/financial-integration/v4.financial-integration.markdown [hunk 5]
```diff
@@ -429,24 +607,44 @@ Code|Description|Category
...
 ---|---|---|---
-`fIDocuments`|`Array`|[Financial document](#schema-fidocument)|The result collection.
-`nextPage`|`string`|-|The URI of the next page of results.
-`numberReturned`|`int32`|-|Number of documents returned in current page.
-`offset`|`string`|-|Next page offset.
+`content`|`Array`|[Financial document](#schema-fidocument)|The result collection.
+`page`|`string`|[Page](#schema-page)|Pagination details
+`links`|`Array`|[Links](#schema-links)|Pagination links.
 
 ### <a name="schema-fidocument"></a>Financial Document
```
Comment (sarraloew) [inline src/api-reference/financial-integration/v4.financial-integration.markdown]: Call this schema FIDocument

File: src/api-reference/financial-integration/v4.financial-integration.markdown [hunk 6]
```diff
@@ -401,6 +462,123 @@ Content-Length: 405
...
+`confirmationRequest`||[body](#confirmationrequest)|**Required** The JSON request to be posted
+
+#### Headers
+
+* [RFC 7235 Authorization](https://tools.ietf.org/html/rfc7235#section-4.2) - Bearer Token that identifies the caller. This is the Company access token.
+* [RFC 7231 Content-Type](https://tools.ietf.org/html/rfc7231#section-3.1.1.5)
+
+#### Payload
+
+[Confirmation Request](schema-paymentrequest)
```
Comment (sarraloew) [inline src/api-reference/financial-integration/v4.financial-integration.markdown]: Swagger doesn't show any content in this request, only a response. I suspect that the markdown is correct, but need your confirmation.

File: src/api-reference/financial-integration/v4.financial-integration.markdown [hunk 7]
```diff
@@ -429,24 +607,44 @@ Code|Description|Category
```
Comment (sarraloew) [inline src/api-reference/financial-integration/v4.financial-integration.markdown:608]: The schema names should match the names in Swagger- people will be looking for the exact strings returned by the API. So, these schemas should be renamed to match what is in Swagger.

**Review by sarraloew — CHANGES_REQUESTED** [review]

File: src/api-reference/financial-integration/v4.financial-integration.markdown (hunk 6 above)
Comment (khoibrandon) [reply src/api-reference/financial-integration/v4.financial-integration.markdown]: true, swagger doesn't show content in request for posting APIs.

File: src/api-reference/financial-integration/v4.financial-integration.markdown (hunk 7 above)
Comment (khoibrandon) [reply src/api-reference/financial-integration/v4.financial-integration.markdown:608]: We were trying to see how to find the difference and unable to see what's wrong here. Can you please give one example.

File: src/api-reference/financial-integration/v4.financial-integration.markdown (hunk 3 above)
Comment (khoibrandon) [reply src/api-reference/financial-integration/v4.financial-integration.markdown]: done, please check

File: src/api-reference/financial-integration/v4.financial-integration.markdown (hunk 5 above)
Comment (khoibrandon) [reply src/api-reference/financial-integration/v4.financial-integration.markdown]: done, please verify

File: src/api-reference/financial-integration/v4.financial-integration.markdown (hunk 2 above)
Comment (khoibrandon) [reply src/api-reference/financial-integration/v4.financial-integration.markdown]: done, pls check

File: src/api-reference/financial-integration/v4.financial-integration.markdown (hunk 4 above)
Comment (khoibrandon) [reply src/api-reference/financial-integration/v4.financial-integration.markdown]: done

File: src/api-reference/financial-integration/v4.financial-integration.markdown (hunk 1 above)
Comment (khoibrandon) [reply src/api-reference/financial-integration/v4.financial-integration.markdown]: Done.

File: src/api-reference/financial-integration/v4.financial-integration.markdown (hunk 7 above)
Comment (sarraloew) [reply src/api-reference/financial-integration/v4.financial-integration.markdown:608]: Examples sent in email.

File: src/api-reference/financial-integration/v4.financial-integration.markdown (hunk 6 above)
Comment (sarraloew) [reply src/api-reference/financial-integration/v4.financial-integration.markdown]: I've added a comment in the Swagger PR asking for it to be updated with this request info.

File: src/api-reference/financial-integration/v4.financial-integration.markdown (hunk 7 above)
Comment (khoibrandon) [reply src/api-reference/financial-integration/v4.financial-integration.markdown:608]: Updated the schema names to match the swagger.  A new PR is uploaded.

File: src/api-reference/financial-integration/v4.financial-integration.markdown [hunk 8]
```diff
@@ -26,19 +26,26 @@ Below are some benefits for using the Financial Integration service:
...
+  * [AcknowledgeResponse](#schema-AcknowledgeResponse)
+  * [AcknowledgeResponseItem](#schema-AcknowledgeResponseItem)
   * [Confirmation Request](#schema-confirmationrequest)
   * [Posting Confirmation Request Item](#schema-postingrequest)
   * [Posting Documents Details](#schema-postingdocs)
   * [System Messages](#schema-systemmsg)
   * [Posting Confirmation Response](#schema-postingconfirmationresponse)
-  * [Posting Confirmation Response Item](#schema-postingresponseitem)
+  * [PostingConfirmationResponseItem](#schema-PostingConfirmationResponseItem)
+  * [Payment Request](#schema-paymentrequest)
```
Comment (sarraloew) [inline src/api-reference/financial-integration/v4.financial-integration.markdown]: This should be Payment Confirmation Request, but it's not included in the Swagger so I don't know the exact format for the name.

File: src/api-reference/financial-integration/v4.financial-integration.markdown [hunk 9]
```diff
@@ -523,6 +723,66 @@ Name|Type|Format|Description
...
+
+Name|Type|Format|Description
+---|---|---|---
+`PaymentConfirmationResponse`|`array`|[Payment Confirmation Response Item](#paymentresponse)|The JSON response body
+
+#### <a name="schema-PaymentConfirmationResponseItem"></a>PaymentConfirmationResponseItem
+
+Name|Type|Format|Description
+---|---|---|---
+`paymentConfirmationResult`|`string`|-|Payment confirmation result. Supported values are: SUCCESS, SYSTEM_ERROR_OCCURRED, NOT_POSTING_CONFIRMED, DOCUMENT_NOT_FOUND, FAILURE, NOOP
```
Comment (sarraloew) [inline src/api-reference/financial-integration/v4.financial-integration.markdown:780]: The Swagger includes a field named paymentRef here, should that be in the markdown?

File: src/api-reference/financial-integration/v4.financial-integration.markdown [hunk 10]
```diff
@@ -401,6 +463,123 @@ Content-Length: 405
...
+`confirmationRequest`||[body](#confirmationrequest)|**Required** The JSON request to be posted
+
+#### Headers
+
+* [RFC 7235 Authorization](https://tools.ietf.org/html/rfc7235#section-4.2) - Bearer Token that identifies the caller. This is the Company access token.
+* [RFC 7231 Content-Type](https://tools.ietf.org/html/rfc7231#section-3.1.1.5)
+
+#### Payload
+
+[Confirmation Request](schema-paymentrequest)
```
Comment (sarraloew) [inline src/api-reference/financial-integration/v4.financial-integration.markdown]: This link is broken

File: src/api-reference/financial-integration/v4.financial-integration.markdown [hunk 11]
```diff
@@ -429,38 +608,58 @@ Code|Description|Category
...
+`number`|int32|-|Page location for which the result is returned, ex: first page starts with number 0, second page starts with 1.
+
+
+### <a name="schema-Link"></a>Link
+
+Name|Type|Format|Description
+---|---|---|---
+`rel`|`string`|`-`|Relation of link, ex: `first`, `self`, `next`, `last`
+`href`|`string`|-|Complete URL for the paginated link
+`media`|`string`|-|null
```
Comment (sarraloew) [inline src/api-reference/financial-integration/v4.financial-integration.markdown]: These descriptions can't be null, they need descriptive text.

File: src/api-reference/financial-integration/v4.financial-integration.markdown [hunk 12]
```diff
@@ -429,38 +608,58 @@ Code|Description|Category
...
+`title`|`string`|-|null
+`type`|`string`|-|null
+`deprecation`|`string`|-|null
```
Comment (sarraloew) [inline src/api-reference/financial-integration/v4.financial-integration.markdown]: The json example includes a field called hreflang, and it is in the Swagger. It needs to be added here.

File: src/api-reference/financial-integration/v4.financial-integration.markdown [hunk 13]
```diff
@@ -429,38 +608,58 @@ Code|Description|Category
...
 ---|---|---|---
 `id`|`string`|`-`|The unique identifier for the document.
-`docType`|`string`|-|Transaction type. Supported values are: `expense`, `invoice`, or `cashadvance`.
+`docType`|`string`|-|Transaction type. Supported values are: `expense`, `invoice`, `cashadvance`, `payroll`, `obligation`.
 `companyId`|`string`|-|Unique identifier for the company in Concur.
 `entityId`|`string`|-|Unique Identifier for the entity in Concur.
 `companyUuid`|`string`|`UUID`|Unique UUID for the company in Concur.
 `erpSystemId`|`string`|-|The external System ID that processed the document.
 `document`|`Array`|-|The JSON financial document. Review the [FI sample documents](#examples) below
 `docStatus`|`string`|-|The financial document status. Supported values are: `READY`, `ACKNOWLEDGED`, `POSTING_CONFIRMED_SUCCESS`, `POSTING_CONFIRMED_FAILURE`
```
Comment (sarraloew) [inline src/api-reference/financial-integration/v4.financial-integration.markdown:626]: The json example includes a links array in the content object. It is not included in the Swagger model for FIDocument. Which one is correct?

**Review by sarraloew — CHANGES_REQUESTED** [review]
Need answers to a few more items.

File: src/api-reference/financial-integration/v4.financial-integration.markdown (hunk 8 above)
Comment (khoibrandon) [reply src/api-reference/financial-integration/v4.financial-integration.markdown]: edited. please check if this is what you actually meant.

File: src/api-reference/financial-integration/v4.financial-integration.markdown (hunk 9 above)
Comment (khoibrandon) [reply src/api-reference/financial-integration/v4.financial-integration.markdown:780]: done.

File: src/api-reference/financial-integration/v4.financial-integration.markdown (hunk 10 above)
Comment (khoibrandon) [reply src/api-reference/financial-integration/v4.financial-integration.markdown]: done

File: src/api-reference/financial-integration/v4.financial-integration.markdown (hunk 11 above)
Comment (khoibrandon) [reply src/api-reference/financial-integration/v4.financial-integration.markdown]: done

File: src/api-reference/financial-integration/v4.financial-integration.markdown (hunk 12 above)
Comment (khoibrandon) [reply src/api-reference/financial-integration/v4.financial-integration.markdown]: added

File: src/api-reference/financial-integration/v4.financial-integration.markdown (hunk 13 above)
Comment (khoibrandon) [reply src/api-reference/financial-integration/v4.financial-integration.markdown:626]: the links array in swagger exists in Resource.FIDocument which is the model that the api shows in swagger. updated the same in markdonw to now refer to schema- Resource.FIDocument.

File: src/api-reference/financial-integration/v4.financial-integration.markdown [hunk 14]
```diff
@@ -88,16 +95,16 @@ https://{datacenterURI}/financialintegration/fi/v4/companies/transactiontypes/{d
 
 Name|Type|Format|Description
 ---|---|---|---
-`docType`|`string`|`path`|**Required** The financial document type to return. Only one type of transaction can be retrieved at a time. Supported values are: `expense`, `invoice`, `cashadvance`.
-`offset`|`string`|`query`|Starting page offset.
+`docType`|`string`|`path`|**Required** The financial document type to return. Only one type of transaction can be retrieved at a time. Supported values are: `expense`, `invoice`, `cashadvance`, `payroll`, `obligation`.
```
Comment (sarraloew) [inline src/api-reference/financial-integration/v4.financial-integration.markdown:98]: Should the docType supported values be the same for all endpoints? Right now two are showing a list of five values, and two are showing a list of three supported values.

Comment (khoibrandon) [reply src/api-reference/financial-integration/v4.financial-integration.markdown:98]: Changes made.  Please review.

**Comment by sarraloew** [comment]
Sent question to Tuan and Srikanth about the messageLanguage supported values.

**Review by sarraloew — APPROVED** [review]

**Comment by sarraloew** [comment]
:shipit:

**Pull request merged**